api/helpers.go

```diff
@@ -0,0 +1,374 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/coreos/go-systemd/dbus"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+)
+
+// dcosTools is implementation of DCOSHelper interface.
+type dcosTools struct {
```

Comment:
Nit: Instead of `dcosTools` maybe make this interface implementation more clear by calling it a `DCOSHost`? 

Also, `DCOS` not `dcos` per lint guidelines.